src/main.rs

```diff
@@ -152,9 +151,15 @@ pub struct Args {
     /// Path to the nvim binary
     pub nvim_bin_path: Option<String>,
 
+    #[cfg(unix)]
+    #[arg(long)]
+    /// Nvim server to connect to (TCP address or Unix socket)
+    pub server: Option<String>,
```

Comment:
I think a nicer way we could handle this would likely be to introduce a new type with `enum`  and implement `ToString` for it just like we do with `TimeoutDuration`. Maybe something like:

```
pub enum NvimTransport {
  SocketAddr(SocketAddr),
  #[cfg(unix)]
  UnixPath(std::path::PathBuf),
}
```

That way we can replace `TcpConnectError` with a single `TransportConnectError` (feel free to come up with a better name) that just holds a `NvimTransport`, and de-duplicate pretty much all of the remote connection error handling by using that through the codebase.

Reply:
I also thought of this, but I liked that the way I have it now I could keep the old documentation that only TCP is supported.

Reply:
But of course, we can still use your solution and have another annotation for the alternative docstring